code/modules/antagonists/heretic/items/corrupted_organs.dm

```diff
@@ -0,0 +1,239 @@
+/// Renders you unable to see people who were heretics at the time that this organ is gained
+/obj/item/organ/internal/eyes/corrupt
+	name = "corrupt orbs"
+	desc = "These eyes have seen something they shouldn't have."
+	/// The override images we are applying
+	var/list/hallucinations
+
+/obj/item/organ/internal/eyes/corrupt/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/corrupted_organ)
+	AddElement(/datum/element/noticable_organ, "%PRONOUN_Their eyes have wide dilated pupils, and no iris. Something is moving in the darkness.", BODY_ZONE_PRECISE_EYES)
+
+/obj/item/organ/internal/eyes/corrupt/on_mob_insert(mob/living/carbon/organ_owner, special, movement_flags)
+	. = ..()
+	if (!organ_owner.client)
+		return
+
+	var/list/human_mobs = GLOB.human_list.Copy()
+	human_mobs -= organ_owner
+	for (var/mob/living/carbon/human/check_human as anything in human_mobs)
+		if (!IS_HERETIC(check_human) && !prob(5)) // Throw in some false positives
+			continue
+		var/image/invisible_man = image('icons/blanks/32x32.dmi', check_human, "nothing")
+		invisible_man.override = TRUE
+		LAZYADD(hallucinations, invisible_man)
+
+	if (LAZYLEN(hallucinations))
+		organ_owner.client.images |= hallucinations
+
+/obj/item/organ/internal/eyes/corrupt/on_mob_remove(mob/living/carbon/organ_owner, special)
+	. = ..()
+	if (!LAZYLEN(hallucinations))
+		return
+	organ_owner.client?.images -= hallucinations
+	QDEL_NULL(hallucinations)
+
+
+/// Sometimes speak in incomprehensible tongues
+/obj/item/organ/internal/tongue/corrupt
+	name = "corrupt tongue"
+	desc = "This one tells only lies."
+
+/obj/item/organ/internal/tongue/corrupt/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/corrupted_organ)
+	AddElement(/datum/element/noticable_organ, "The inside of %PRONOUN_Their mouth is full of stars.", BODY_ZONE_PRECISE_MOUTH)
+
+/obj/item/organ/internal/tongue/corrupt/on_mob_insert(mob/living/carbon/organ_owner, special, movement_flags)
+	. = ..()
+	RegisterSignal(organ_owner, COMSIG_MOB_SAY, PROC_REF(on_spoken))
+
+/obj/item/organ/internal/tongue/corrupt/on_mob_remove(mob/living/carbon/organ_owner, special)
+	. = ..()
+	UnregisterSignal(organ_owner, COMSIG_MOB_SAY)
+
+/// When the mob speaks, sometimes put it in a different language
+/obj/item/organ/internal/tongue/corrupt/proc/on_spoken(mob/living/organ_owner, list/speech_args)
+	SIGNAL_HANDLER
+	if (organ_owner.has_reagent(/datum/reagent/water/holywater) || prob(60))
+		return
+	speech_args[SPEECH_LANGUAGE] = /datum/language/shadowtongue
+
+
+/// Randomly secretes alcohol or hallucinogens when you're drinking something
+/obj/item/organ/internal/liver/corrupt
+	name = "corrupt liver"
+	desc = "After what you've seen you could really go for a drink."
+	/// How much extra ingredients to add?
+	var/amount_added = 5
+	/// What extra ingredients can we add?
+	var/list/extra_ingredients = list(
```

Comment:
I don't think this one makes much sense. Why does being cursed by the eldritch gates of the Mansus make you an alcoholic/druggie? Did you run out of ideas?
If so, my suggestions would be 'occasionally turn a reagent in the bloodstream to its inverse', 'sets reagent purity to 1%', or 'eldritch essence manifests itself in you occassionally'

Reply:
Honestly it's mostly just because I like the things that it does to your screen and don't want to extract all that code into external procs when I could just put the reagents in you.
Also I think it's funny.

And I like it having a specific trigger more than it sort of "just happening", I already have a lot of "it just sort of happens" effects in here.

Reply:
I'm not totally averse to it doing something else but I don't like those ideas more than what I already have. I'll sleep on it.